source/gui/utilityDialogs.py

```diff
@@ -0,0 +1,58 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2024 NV Access Limited, Luke Davis
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import wx
+
+from .nvdaControls import MessageDialog
+
+
+class ContinueCancelDialog(MessageDialog):
+	"""
+	This implementation of a `gui.nvdaControls.MessageDialog`, provides `Continue` and `Cancel` buttons as its controls.
+	These serve the same functions as `OK` and `Cancel` in other dialogs, but may be more desirable in some situations.
+	"""
+
+	def __init__(
+		self,
+		parent: wx.Frame,
+		title: str,
+		message: str,
+		dialogType: int = MessageDialog.DIALOG_TYPE_STANDARD,
+		continueButtonFirst: bool = True,
+	) -> None:
+		"""Creates a ContinueCancelDialog MessageDialog.
+
+		:param parent: The parent window for the dialog, usually `gui.mainFrame`.
+		:param title: The title or caption of the dialog.
+		:param message: The message to be shown in the dialog.
+		:param dialogType: One of the dialog type constants from MessageDialog, defaults to standard.
+		:param continueButtonFirst: If True, the Continue button will appear first, and be selected when the dialog
```

Comment:
I did it this way because I couldn't find another way. Originally I tried fixed positions for the two buttons, and just using `SetDefault()` on the `Cancel` button, if the caller asked for that. However I could never get `SetDefault` to be honored in the overridden `_addButtons` method, and could not figure out why it was being ignored.

I found, however, that whichever button was added first, became the default. Since I wanted this dialog to be somewhat generic, I switched to doing this. I put a comment to that effect in `_addButtons`.